jax/numpy/lax_numpy.py

```diff
@@ -596,6 +598,34 @@ def round(a, decimals=0):
 around = round
 
 
+# Caution: If fast math mode is enabled, the semantics of inf and nan are not
+# preserved by XLA/LLVM, and the behavior of inf/nan values is unpredictable.
+# To disable fast math mode on CPU, set the environment variable
+# XLA_FLAGS=--xla_cpu_enable_fast_math=false.
+
+def isfinite(x):
+  dtype = _dtype(x)
+  if issubdtype(dtype, floating):
+    return lax.is_finite(x)
+  elif issubdtype(dtype, complexfloating):
+    return lax.bitwise_and(lax.is_finite(real(x)), lax.is_finite(imag(x)))
+  else:
+    return full_like(x, True, dtype=bool_)
+
+def isinf(x):
+  dtype = _dtype(x)
+  if issubdtype(dtype, floating):
+    return lax.eq(lax.abs(x), inf)
+  elif issubdtype(dtype, complexfloating):
+    return lax.bitwise_or(lax.eq(lax.abs(real(x)), inf),
+                          lax.eq(lax.abs(imag(x)), inf))
+  else:
+    return full_like(x, False, dtype=bool_)
+
+def isnan(x):
+  return bitwise_and(bitwise_not(isfinite(x)), bitwise_not(isinf(x)))
```

Comment:
Could call `lax.bitwise_and` and `lax.bitwise_not` to avoid one level of wrapping/indirection here.